src/MainView.vala

```diff
@@ -67,17 +74,64 @@ public class Network.MainView : Gtk.Paned {
         sidebar.add (scrolled_window);
         sidebar.add (footer);
 
-        position = 240;
-        pack1 (sidebar, false, false);
-        pack2 (content, true, false);
+#if USE_UBUNTU_SYSTEM_SERVICE
+        if (proxy_permission == null) {
+            try {
+                proxy_permission = new Polkit.Permission.sync (
+                    "io.elementary.switchboard.network.setproxy",
+                    new Polkit.UnixProcess (Posix.getpid ())
+                );
+            } catch (Error e) {
+                warning ("Unable to create polkit permission object: %s", e.message);
+            }
+        }
+
+        if (proxy_permission != null) {
+            var proxy_permission_infobar = new Gtk.InfoBar ();
+            proxy_permission_infobar.message_type = Gtk.MessageType.INFO;
+
+            var infobar_action_area = (proxy_permission_infobar.get_action_area () as Gtk.Container);
+            infobar_action_area.add (new Gtk.LockButton (proxy_permission));
+
+            var infobar_content_area = (proxy_permission_infobar.get_content_area () as Gtk.Container);
+            infobar_content_area.add (new Gtk.Label (_("Administrator rights are required to set a system-wide proxy")));
+
+            proxy_permission_revealer = new Gtk.Revealer ();
+            proxy_permission_revealer.add (proxy_permission_infobar);
+
+            proxy_permission.notify["allowed"].connect (() => {
+                proxy_permission_revealer.reveal_child = !proxy_permission.allowed;
+                if (page is Widgets.ProxyPage) {
+                    (page as Widgets.ProxyPage).systemwide_available = proxy_permission.allowed;
```

Comment:
At this point we should explicitly get the proxy page from `DeviceList` and set it without depending on the fact that current page is the proxy page.